lib/swaps/Swaps.ts

```diff
@@ -565,7 +603,11 @@ class Swaps extends EventEmitter {
       await takerSwapClient.addInvoice(deal.rHash, deal.takerAmount);
     } catch (err) {
       this.failDeal(deal, SwapFailureReason.UnexpectedClientError, err.message);
-      await this.sendErrorToPeer(peer, rHash, err.message);
+      await this.sendErrorToPeer({
+        peer,
+        rHash,
+        errorMessage: err.message,
```

Comment:
I think we can set the failure reason to `SwapFailureReason.UnexpectedClientError` here.